src/Castle.Windsor/MicroKernel/Registration/BasedOnDescriptor.cs

```diff
@@ -535,11 +536,14 @@ private static bool IsBasedOnGenericInterface(Type type, Type basedOn, out Type[
 			var types = new List<Type>(4);
 			foreach (var @interface in type.GetInterfaces())
 			{
-				if (@interface.IsGenericType &&
+				if (@interface.GetTypeInfo().IsGenericType &&
 				    @interface.GetGenericTypeDefinition() == basedOn)
 				{
-					if (@interface.ReflectedType == null &&
-					    @interface.ContainsGenericParameters)
+					if (@interface.GetTypeInfo().ContainsGenericParameters
+#if FEATURE_REFLECTION_REFLECTEDTYPE
+						&& @interface.ReflectedType == null
+#endif
```

Comment:
Sounds good to me. I think the only scenario where `ReflectedType` would return null would be for global methods on a module, and `DeclaringType` would also be null.